scripts/globals/crafting/crafting_image_support.lua

```diff
@@ -114,10 +115,27 @@ xi.crafting.oldImageSupportOnEventFinish = function(player, csid, option, npc)
     local skillId          = npcTable[npcName][4]
     local effectId         = npcTable[npcName][5]
     local gilCost          = 0
+    local joinedGuildMask  = player:getCharVar('Guild_Member')
 
-    -- Calculate Gil cost.
+    -- Handle advance support.
     if messageParameter == 0 then
+        -- Calculate Gil cost.
         gilCost = (player:getSkillRank(skillId) + 1) * 30
+
+        -- Handle special initial messages.
+        if
+            guildId == xi.guild.GOLDSMITHING and                          -- Goldsmithing guild.
+            utils.mask.getBit(joinedGuildMask, xi.guild.GOLDSMITHING) and -- Has joined goldsnmithing guild.
+            not utils.mask.getBit(joinedGuildMask, 24)                    -- Has not spoken to npc after joining guild.
+        then
+            player:incrementCharVar('Guild_Member', bit.lshift(1, 24))
```

Comment:
Not a fan of forever charvars, and guild membership could most likely move to a set of uniques in the future.  This is also a bit dangerous to add the bit if anyone follows this pattern without checking first (no issue here though)